utils/misc.ml

```diff
@@ -684,3 +684,36 @@ let delete_eol_spaces src =
   in
   let stop = loop 0 0 in
   Bytes.sub_string dst 0 stop
+
+exception HookExit of exn
```

Comment:
What the rationale for this HookExit, and the fact that any other exception terminates the compiler (or toplevel) immediately.  I'd rather have `fold_hooks` wrap the exception into another constructor that records the hook name, and let that new exception propagates as usual.